nixos/tests/systemd-networkd-ipv6-prefix-delegation.nix

```diff
@@ -0,0 +1,298 @@
+              # prefix or not.  Note: On some installation it makes sense to
+              # always force that option on the DHPCv6 server since there are
+              # certain CPEs that are just not setting this field but happily
+              # accept the delegated prefix.
+              PrefixDelegationHint  = "::/48";
+            };
+            ipv6PrefixDelegationConfig = {
+              # Let networkd know that we would very much like to use DHCPv6
+              # to obtain the "managed" information. Not sure why they can't
+              # just take that from the upstream RAs.
+              Managed = true;
+            };
+          };
+
+          # Interface to the client. Here we should redistribute a /64 from
+          # the prefix we received from the ISP.
+          "01-eth2" = {
+            name = "eth2";
+            networkConfig = {
+              Description = "Client interface";
+              # the client shouldn't be allowed to send us RAs, that would be weird.
+              IPv6AcceptRA = false;
+
+              # Just delegate prefixes from the DHCPv6 PD pool.
+              # If you also want to distribute a local ULA prefix you want to
+              # set this to `yes` as that includes both static prefixes as well
+              # as PD prefixes.
+              IPv6PrefixDelegation = "dhcpv6";
+            };
+            # finally "act as router" (according to systemd.network(5))
+            ipv6PrefixDelegationConfig = {
+              RouterLifetimeSec = 300; # required as otherwise no RA's are being emitted
+
+              # In a production environment you should consider setting these as well:
+              #EmitDNS = true;
+              #EmitDomains = true;
+              #DNS= = "fe80::1"; # or whatever "well known" IP your router will have on the inside.
+            };
+
+            # This adds a "random" ULA prefix to the interface that is being
+            # advertised to the clients.
+            # Not used in this test (besides adding an address of another scope
+            # to the interaces).
+            ipv6Prefixes = [
+              {
+                ipv6PrefixConfig = {
+                  AddressAutoconfiguration = true;
+                  PreferredLifetimeSec = 1800;
+                  ValidLifetimeSec = 1800;
+                };
+              }
+            ];
+          };
+
+          # finally we are going to add a static IPv6 unique local address to
+          # the "lo" interface.  This will serve as ICMPv6 echo target to
+          # verify connectivity from the client to the router.
+          "01-lo" = {
+            name = "lo";
+            addresses = [
+              { addressConfig.Address = "FD42::1/128"; }
+            ];
+          };
+        };
+      };
+
+      # make the network-online target a requirement, we wait for it in our test script
+      systemd.targets.network-online.wantedBy = [ "multi-user.target" ];
```

Comment:
Hrm, I don't see us waiting for `router`'s `multi-user.target` anywhere in the test script, we wait for `systemd-networkd`, which should already be part of `multi-user.target` - we might just want to drop this line, and wait for `router` to reach `systemd-networkd.service`, and later `network-online.target` in the test script.

Reply:
hah, `network-online.target` is not being pulled in by anything since #86273. I expect a bunch of tests to be broken now. I had to change this just after I rebased.

Pulling this in is probably the easiest way to still have a way to distinguish between "multi-user.target" reached and `networkd` telling is that the setup is complete.

Reply:
But wait, shouldn't the `network-online.target` still be reached as soon as networkd determined it's online, even if there's no unit having it as `Wants=`?

Reply:
No, the `network-online.target` spawns a service that waits for networkd/or whatever to signal (via dbus) that the system is now online and block the target goal until then.